src/main/scala/mesosphere/marathon/scheduling/LegacyScheduler.scala

```diff
@@ -0,0 +1,116 @@
+
+  // We cannot process more Add/Update requests for one runSpec in parallel because it leads to race condition.
+  // See MARATHON-8320 for details. The queue handling is helping us ensure we change one instance at a time.
+  private[this] val serializeUpdates: WorkQueue = WorkQueue("LegacyScheduler", maxConcurrent = 1, maxQueueLength = Int.MaxValue)
+
+  override def schedule(runSpec: RunSpec, count: Int)(implicit ec: ExecutionContext): Future[Done] = serializeUpdates {
+    async {
+      val instancesToSchedule = 0.until(count).map { _ => Instance.scheduled(runSpec, Instance.Id.forRunSpec(runSpec.id)) }
+      await(instanceTracker.schedule(instancesToSchedule))
+
+      // Let launch queue manage task launcher actor.
+      await(launchQueue.sync(runSpec))
+
+      Done
+    }
+  }
+
+  override def reschedule(instance: Instance, runSpec: RunSpec)(implicit ec: ExecutionContext): Future[Done] = serializeUpdates {
+    async {
+      /* This method is actually and update from Goal.Stopped to Goal.Running. However, only instances with reservations
+         support such an update. MARATHON-8373 will introduce incarnations for ephemeral instances and enable rescheduling
+         support for all instances.
+       */
+      assert(instance.isReserved && instance.state.goal == Goal.Stopped)
+      await(instanceTracker.process(RescheduleReserved(instance, runSpec.version)))
+
+      // Let launch queue manage task launcher actor.
+      await(launchQueue.sync(runSpec))
+
+      Done
+    }
+  }
+
+  // TODO(karsten): Investigate how we can drop this method as it leaks implementation details of the scheduler.
+  override def resetDelay(spec: RunSpec): Unit = launchQueue.resetDelay(spec)
+
+  /* The sync is responsible for two things:
+     1. Tell the [[LaunchQueueActor]] to start a [[TaskLauncherActor]].
+     2. Tell the [[TaskLauncherActor]] about the new run spec version to start.
+
+     Once we attach the run spec to an instance we can drop 2. This is addressed in MARATHON-8325.
+     Once the [[TaskLauncherActor]] is just a pure function called un [[OfferMatcherManagerActor]] we can drop 1. and
+     thus this method altogether. See MARATHON-8444 for that.
+
+     Overall this method should not be part of the future interface as it leaks implementation details of the scheduler
+     internals.
+   */
+  override def sync(spec: RunSpec)(implicit ec: ExecutionContext): Future[Done] = serializeUpdates { launchQueue.sync(spec) }
+
+  override def getInstances(runSpecId: PathId)(implicit ec: ExecutionContext): Future[Seq[Instance]] = serializeUpdates {
+    instanceTracker.specInstances(runSpecId)
+  }
+
+  override def getInstance(instanceId: Instance.Id)(implicit ec: ExecutionContext): Future[Option[Instance]] = serializeUpdates {
+    instanceTracker.get(instanceId)
+  }
+
+  @SuppressWarnings(Array("all")) // async/await
+  override def run(instances: Seq[Instance])(implicit ec: ExecutionContext): Future[Done] = serializeUpdates {
```

Comment:
I think we can drop this method because this operation is not supported. It's only supported by resident tasks and those have that handled differently